File: fixtures/python/calculator.py
def calculate_total(items):
    """Sum the price of all items."""
    total = 0
    for item in items:
        total += item["price"]
    return total


def apply_discount(total, percent):
    """Apply a percentage discount to a total."""
    if percent < 0 or percent > 100:
        raise ValueError("percent must be 0-100")
    return total * (1 - percent / 100)

File: fixtures/go/calculator.go
package calculator

// CalculateTotal sums the price of all items.
func CalculateTotal(prices []float64) float64 {
	total := 0.0
	for _, p := range prices {
		total += p
	}
	return total
}

// ApplyDiscount applies a percentage discount to a total.
func ApplyDiscount(total, percent float64) float64 {
	if percent < 0 || percent > 100 {
		panic("percent must be 0-100")
	}
	return total * (1 - percent/100)
}
